### ISSUE TITLE
Params extension does not work when param_type is body

### ISSUE THREAD

**Issue by lenon** [ruby-grape/grape-swagger#901]
Hi!

When I try to document an extension for a parameter like this:

```ruby
requires :foo, type: String, documentation: { x: { some: 'stuff' }, param_type: 'body' }
```

It does not work. The documentation is generated without the `x-some` extension.

But when `param_type` is omitted:

```ruby
requires :foo, type: String, documentation: { x: { some: 'stuff' } }
```

The generated documentation includes the `x-some` extension.

I wrote the following script to help reproduce this problem:

```ruby
require 'grape'
require 'grape-swagger'

class API < Grape::API
  format :json

  params do
    requires :foo, type: String, documentation: { x: { some: 'stuff' } }
  end
  post :test1 do
  end

  params do
    requires :foo, type: String, documentation: { x: { some: 'stuff' }, param_type: 'body' }
  end
  post :test2 do
  end

  add_swagger_documentation
end
```

This is the generated documentation:

(I redacted it to keep only the relevant content)

```json
{
    ...
    "paths": {
        "/test1": {
            "post": {
                ...
                "parameters": [
                    {
                        "in": "formData",
                        "name": "foo",
                        "type": "string",
                        "required": true,
                        "x-some": "stuff"
                    }
                ],
                ...
            }
        },
        "/test2": {
            "post": {
                ...
                "parameters": [
                    {
                        "name": "postTest2",
                        "in": "body",
                        "required": true,
                        "schema": {
                            "$ref": "#/definitions/postTest2"
                        }
                    }
                ],
                ...
            }
        }
    },
    "definitions": {
        "postTest2": {
            "type": "object",
            "properties": {
                "foo": {
                    "type": "string"
                }
            },
            "required": [
                "foo"
            ]
        }
    }
}
```

As you can see in the JSON, when `in` is `formData`, `"x-some": "stuff"` is present. When `in` is `body`, it isn't.

I think this happens because of this method in the `MoveParams` class, which appears to be used to document body params:

https://github.com/ruby-grape/grape-swagger/blob/3a5da56fb3ad19dcb07cf3ef86549bb811e22dcc/lib/grape-swagger/doc_methods/move_params.rb#L104-L115

It uses a fixed list of properties from `property_keys` and documents them if they are present. Anything else that is not in this list gets discarded.

Maybe it would be possible to change this method to check for keys that start with `x-` and document them.

Does that make sense? I can try to make a PR if it does.

Thanks!

**Comment by LeFnord** [comment]
Hi @lenon … thanks for the detailed bug report, and yes I PR would be very welcome … 👍

**Comment by lenon** [comment]
Fixed by #918. Thanks @numbata 😃
